src/loom-provider.ts

```diff
@@ -572,7 +573,9 @@ export class LoomProvider {
   }
 
   private _netVersion() {
-    return this._client.chainId
+    // Truffle don't support utf-8 values here
+    // So the chainId is converted to hex value
+    return new BN(utf8ToHex(this._client.chainId)).toString(10)
```

Comment:
I suspect that if the BN can't fit the hex value into a 53-bit integer it will throw an exception here.

Reply:
It works because it returns a `string` like:

```js
const utf8ToHex = (str) => '0x' + Buffer.from(str, 'utf-8').toString('hex')
new BN(utf8ToHex('ThisIsMyCoolDappChainIdIHaveChooseThisIdBecauseIsSuperCool')).toString(10)
// '3354686973497353794375757244617070436861697449644948617665436875757365546869734964426563617573654973537570657243757572'
```

Reply:
Actually I think in this case is better to return the `ASCII` instead of using BN

Reply:
Ah, OK, if Truffle can handle that it's fine I guess.